src/phoebus_driver.cpp

```diff
@@ -130,27 +129,8 @@ void PhoebusDriver::PostInitializationCommunication() {
     const auto local = parthenon::BoundaryType::local;
     const auto nonlocal = parthenon::BoundaryType::nonlocal;
 
-    auto start_recv =
-        tl.AddTask(none, parthenon::cell_centered_bvars::StartReceiveBoundBufs<any>, md);
-
-    auto send = tl.AddTask(start_recv,
-                           parthenon::cell_centered_bvars::SendBoundBufs<nonlocal>, md);
-
-    auto send_local =
-        tl.AddTask(start_recv, parthenon::cell_centered_bvars::SendBoundBufs<local>, md);
-    auto recv_local = tl.AddTask(
-        start_recv, parthenon::cell_centered_bvars::ReceiveBoundBufs<local>, md);
-    auto set_local =
-        tl.AddTask(recv_local, parthenon::cell_centered_bvars::SetBounds<local>, md);
-
-    auto recv = tl.AddTask(
-        start_recv, parthenon::cell_centered_bvars::ReceiveBoundBufs<nonlocal>, md);
-    auto set = tl.AddTask(recv, parthenon::cell_centered_bvars::SetBounds<nonlocal>, md);
-
-    if (pmesh->multilevel) {
-      tl.AddTask(set | set_local, parthenon::cell_centered_bvars::RestrictGhostHalos, md,
-                 false);
-    }
+    auto boundary_tasks =
+        parthenon::AddBoundaryExchangeTasks(none, tl, md, pmesh->multilevel);
```

Comment:
@brryan and or @lroberts36 I'd like your eyes on the changes to the task list, in case I missed something.

Reply:
This looks good to me (it's a very nice cleanup of the driver code actually) except I'm not familiar with `parthenon::AddBoundaryExchangeTasks` so maybe worth @lroberts36 taking a look as well.